[connector/spanmetrics] Test TestConnectorConsumeTracesExpiredMetrics fails intermittently on actuated ARM runners

### Component(s)

connector/spanmetrics

### Describe the issue you're reporting

[CI/CD failure link](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8694589952/job/23843888472)

Failure output:
```
Running target 'test-with-cover' in module 'connector/spanmetricsconnector' as part of group 'connector'
make --no-print-directory -C connector/spanmetricsconnector test-with-cover
mkdir -p /home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/coverage/unit
/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/gotestsum --rerun-fails=1 --packages="./..." -- -race -timeout 300s -parallel 4 --tags="" -cover -covermode=atomic -args -test.gocoverdir="/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/coverage/unit"
∅  internal/metadata
✖  . (327ms)
✓  internal/cache (1.033s) (coverage: 73.1% of statements)
✓  internal/metrics (1.049s) (coverage: 35.4% of statements)
ERROR rerun aborted because previous run had a suspected panic and some test may not have run

=== Failed
=== FAIL: . TestConnectorConsumeTracesExpiredMetrics (0.01s)
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 1091 [running]:
testing.tRunner.func1.2({0xafc560, 0xc00040c090})
	/home/runner/actions-runner/_work/_tool/go/1.21.9/arm64/src/testing/testing.go:1545 +0x2c8
testing.tRunner.func1()
	/home/runner/actions-runner/_work/_tool/go/1.21.9/arm64/src/testing/testing.go:1548 +0x4c4
panic({0xafc560?, 0xc00040c090?})
	/home/runner/actions-runner/_work/_tool/go/1.21.9/arm64/src/runtime/panic.go:920 +0x26c
github.com/open-telemetry/opentelemetry-collector-contrib/connector/spanmetricsconnector.TestConnectorConsumeTracesExpiredMetrics(0xc00040[89](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8694589952/job/23843888472#step:9:90)c0)
	/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/connector/spanmetricsconnector/connector_test.go:1301 +0xcfc
testing.tRunner(0xc0004089c0, 0xbad8a0)
	/home/runner/actions-runner/_work/_tool/go/1.21.9/arm64/src/testing/testing.go:1595 +0x1b4
created by testing.(*T).Run in goroutine 1
	/home/runner/actions-runner/_work/_tool/go/1.21.9/arm64/src/testing/testing.go:1648 +0x5ec

=== FAIL: . TestLoadConfig (unknown)

=== FAIL: . TestConsumeTraces (unknown)

DONE 80 tests, 3 failures in 3.249s
make[2]: *** [../../Makefile.Common:131: test-with-cover] Error 3
make[1]: *** [Makefile:165: connector/spanmetricsconnector] Error 2
make: *** [Makefile:117: gotest-with-cover] Error 2
make[1]: Leaving directory '/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
```

--- a/.github/workflows/build-and-test-arm.yml
+++ b/.github/workflows/build-and-test-arm.yml
@@ -14,7 +14,6 @@ env:
   # Make sure to exit early if cache segment download times out after 2 minutes.
   # We limit cache download as a whole to 5 minutes.
   SEGMENT_DOWNLOAD_TIMEOUT_MINS: 2
-  GOPROXY: https://goproxy1.cncf.selfactuated.dev,direct
 
 # Do not cancel this workflow on main. See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16616
 concurrency:

--- a/.github/workflows/build-and-test-arm.yml
+++ b/.github/workflows/build-and-test-arm.yml
@@ -46,7 +46,7 @@ jobs:
           - cmd-1
           - other
     timeout-minutes: 30
-    runs-on: actuated-arm64-4cpu-4gb
+    runs-on: macos-14
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
@@ -72,7 +72,7 @@ jobs:
         run: make -j2 gotest GROUP=${{ matrix.group }}
   arm-unittest:
     if: ${{ github.actor != 'dependabot[bot]' && (contains(github.event.pull_request.labels.*.name, 'Run ARM') || github.event_name == 'push' || github.event_name == 'merge_group') }}
-    runs-on: actuated-arm64-4cpu-4gb
+    runs-on: macos-14
     needs: [arm-unittest-matrix]
     steps:
       - name: Print result

--- a/.github/workflows/build-and-test-arm.yml
+++ b/.github/workflows/build-and-test-arm.yml
@@ -26,6 +26,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
+        os: [otel-linux-arm64, macos-14]
         group:
           - receiver-0
           - receiver-1
@@ -45,7 +46,7 @@ jobs:
           - cmd-1
           - other
     timeout-minutes: 30
-    runs-on: macos-14
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
@@ -71,7 +72,7 @@ jobs:
         run: make -j2 gotest GROUP=${{ matrix.group }}
   arm-unittest:
     if: ${{ github.actor != 'dependabot[bot]' && (contains(github.event.pull_request.labels.*.name, 'Run ARM') || github.event_name == 'push' || github.event_name == 'merge_group') }}
-    runs-on: macos-14
+    runs-on: ubuntu-latest
     needs: [arm-unittest-matrix]
     steps:
       - name: Print result

--- a/extension/bearertokenauthextension/bearertokenauth_test.go
+++ b/extension/bearertokenauthextension/bearertokenauth_test.go
@@ -8,6 +8,7 @@ import (
 	"fmt"
 	"net/http"
 	"os"
+	"path/filepath"
 	"testing"
 	"time"
 
@@ -102,7 +103,7 @@ func TestBearerAuthenticator(t *testing.T) {
 
 func TestBearerStartWatchStop(t *testing.T) {
 	cfg := createDefaultConfig().(*Config)
-	cfg.Filename = "test.token"
+	cfg.Filename = filepath.Join("testdata", t.Name()+".token")
 
 	bauth := newBearerTokenAuth(cfg, zaptest.NewLogger(t))
 	assert.NotNil(t, bauth)
@@ -152,7 +153,7 @@ func TestBearerStartWatchStop(t *testing.T) {
 func TestBearerTokenFileContentUpdate(t *testing.T) {
 	scheme := "TestScheme"
 	cfg := createDefaultConfig().(*Config)
-	cfg.Filename = "test.token"
+	cfg.Filename = filepath.Join("testdata", t.Name()+".token")
 	cfg.Scheme = scheme
 
 	bauth := newBearerTokenAuth(cfg, zaptest.NewLogger(t))

--- a/extension/bearertokenauthextension/test.token
+++ b/extension/bearertokenauthextension/test.token
@@ -1,1 +1,0 @@
-eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...+file+

--- a/extension/bearertokenauthextension/testdata/TestBearerStartWatchStop.token
+++ b/extension/bearertokenauthextension/testdata/TestBearerStartWatchStop.token
@@ -1,0 +1,1 @@
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...+file+

--- a/extension/bearertokenauthextension/testdata/TestBearerTokenFileContentUpdate.token
+++ b/extension/bearertokenauthextension/testdata/TestBearerTokenFileContentUpdate.token
@@ -1,0 +1,1 @@
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...+file

--- a/receiver/hostmetricsreceiver/hostmetrics_receiver_test.go
+++ b/receiver/hostmetricsreceiver/hostmetrics_receiver_test.go
@@ -36,7 +36,7 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/processscraper"
 )
 
-var armMetrics = []string{
+var allMetrics = []string{
 	"system.cpu.time",
 	"system.cpu.load_average.1m",
 	"system.cpu.load_average.5m",
@@ -54,11 +54,7 @@ var armMetrics = []string{
 	"system.network.io",
 	"system.network.packets",
 	"system.paging.operations",
-}
-
-var archSpecificMetrics = map[string][]string{
-	"arm64": armMetrics,
-	"amd64": append(armMetrics, "system.paging.usage"),
+	"system.paging.usage",
 }
 
 var resourceMetrics = []string{
@@ -174,7 +170,7 @@ func assertIncludesExpectedMetrics(t *testing.T, got pmetric.Metrics) {
 
 	// verify the expected list of metrics returned (os dependent)
 	var expectedMetrics []string
-	expectedMetrics = append(expectedMetrics, archSpecificMetrics[runtime.GOARCH]...)
+	expectedMetrics = append(expectedMetrics, allMetrics...)
 	expectedMetrics = append(expectedMetrics, systemSpecificMetrics[runtime.GOOS]...)
 	assert.Equal(t, len(expectedMetrics), len(returnedMetrics))
 	for _, expected := range expectedMetrics {

--- a/receiver/hostmetricsreceiver/hostmetrics_receiver_test.go
+++ b/receiver/hostmetricsreceiver/hostmetrics_receiver_test.go
@@ -54,7 +54,6 @@ var allMetrics = []string{
 	"system.network.io",
 	"system.network.packets",
 	"system.paging.operations",
-	"system.paging.usage",
 }
 
 var resourceMetrics = []string{
@@ -168,9 +167,12 @@ func assertIncludesExpectedMetrics(t *testing.T, got pmetric.Metrics) {
 		}
 	}
 
-	// verify the expected list of metrics returned (os dependent)
-	var expectedMetrics []string
-	expectedMetrics = append(expectedMetrics, allMetrics...)
+	// verify the expected list of metrics returned (os/arch dependent)
+	expectedMetrics := allMetrics
+	if !(runtime.GOOS == "linux" && runtime.GOARCH == "arm64") {
+		expectedMetrics = append(expectedMetrics, "system.paging.usage")
+	}
+
 	expectedMetrics = append(expectedMetrics, systemSpecificMetrics[runtime.GOOS]...)
 	assert.Equal(t, len(expectedMetrics), len(returnedMetrics))
 	for _, expected := range expectedMetrics {

--- a/receiver/hostmetricsreceiver/internal/scraper/pagingscraper/paging_scraper_test.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/pagingscraper/paging_scraper_test.go
@@ -82,10 +82,6 @@ func TestScrape(t *testing.T) {
 			if runtime.GOOS == "windows" {
 				expectedMetrics = 3
 			}
-			// ARM runner has no swap:
-			if runtime.GOARCH == "arm64" {
-				expectedMetrics = 2
-			}
 
 			assert.Equal(t, expectedMetrics, md.MetricCount())
 
@@ -100,12 +96,11 @@ func TestScrape(t *testing.T) {
 
 			internal.AssertSameTimeStampForMetrics(t, metrics, 0, metrics.Len()-2)
 			startIndex++
-			if runtime.GOARCH != "arm64" {
-				assertPagingUsageMetricValid(t, metrics.At(startIndex))
-				internal.AssertSameTimeStampForMetrics(t, metrics, startIndex, metrics.Len())
-				startIndex++
-				assertPagingUtilizationMetricValid(t, metrics.At(startIndex))
-			}
+
+			assertPagingUsageMetricValid(t, metrics.At(startIndex))
+			internal.AssertSameTimeStampForMetrics(t, metrics, startIndex, metrics.Len())
+			startIndex++
+			assertPagingUtilizationMetricValid(t, metrics.At(startIndex))
 		})
 	}
 }

--- a/receiver/hostmetricsreceiver/internal/scraper/pagingscraper/paging_scraper_test.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/pagingscraper/paging_scraper_test.go
@@ -83,6 +83,11 @@ func TestScrape(t *testing.T) {
 				expectedMetrics = 3
 			}
 
+			// linux + ARM runner has no swap
+			if runtime.GOOS == "linux" && runtime.GOARCH == "arm64" {
+				expectedMetrics = 2
+			}
+
 			assert.Equal(t, expectedMetrics, md.MetricCount())
 
 			startIndex := 0
@@ -97,10 +102,12 @@ func TestScrape(t *testing.T) {
 			internal.AssertSameTimeStampForMetrics(t, metrics, 0, metrics.Len()-2)
 			startIndex++
 
-			assertPagingUsageMetricValid(t, metrics.At(startIndex))
-			internal.AssertSameTimeStampForMetrics(t, metrics, startIndex, metrics.Len())
-			startIndex++
-			assertPagingUtilizationMetricValid(t, metrics.At(startIndex))
+			if !(runtime.GOOS == "linux" && runtime.GOARCH == "arm64") {
+				assertPagingUsageMetricValid(t, metrics.At(startIndex))
+				internal.AssertSameTimeStampForMetrics(t, metrics, startIndex, metrics.Len())
+				startIndex++
+				assertPagingUtilizationMetricValid(t, metrics.At(startIndex))
+			}
 		})
 	}
 }

